registry/registry.go

```diff
@@ -64,6 +65,11 @@ type Registry interface {
 	// so sync load the instance of the preparing to subscribe service before
 	// formally subscribing.
 	LoadSubscribeInstances(*common.URL, NotifyListener) error
+
+	// SubscribeURL is required to support the contract:
+	// 1. ServiceInstancesChangedListener provide by SubscribeURL
+	// 2. ServiceMappingChangedEvent provide by Subscribe
+	SubscribeURL(*common.URL, NotifyListener, *gxset.HashSet)
```

Comment:
I don't think adding a `SubscribeURL` to the `Registry` iface is a good idea because the `SubscribeURL` is an exclusive behavior of the service discovery registry.

Reply:
so we can export ServiceDiscoveryRegistry struct